src/service/external_service.py

```diff
@@ -86,35 +89,34 @@ def get_spotify_data(user_uuid, app):
                     for line in d:
                         if TrackModel.query.filter_by(spotify_id=line['spotify_id']).first() is None:
                             new_track = TrackModel(
-                                artist_name=" & ".join(line['artist_name']),
+                                artist_name=line['artist_name'],
                                 title=line['title'],
-                                year=line['year'].split("-")[0] if line['year'] is not None else None,
+                                year=line['year'],
                                 release=line['release'],
                                 spotify_id=line['spotify_id'],
                                 covert_art_url=line['cover_art_url']
                             )
                             db.session.add(new_track)
                             db.session.flush()
+                        
                 db.session.commit()
 
                 for d in data:
                     for line in d:
-                        track = TrackModel.query.filter_by(
-                            spotify_id=line['spotify_id']).first()
+                        track = TrackModel.query.filter_by(spotify_id=line['spotify_id']).first()
                         if ((meta := MetaUserTrackModel.query.filter_by(track_id=track.track_id).first()) is None):
                             new_meta_user_track = MetaUserTrackModel(
                                 user_id=user.user_id,
                                 track_id=track.track_id,
                                 play_count=1,
-                                last_played_date=dateutil.parser.isoparse(
-                                    line['played_at'][:-1]) if "played_at" in line.keys() else None
+                                last_played_date=line['played_at']
                             )
                             db.session.add(new_meta_user_track)
                             db.session.flush()
                         else :
                             # ? a verif
```

Comment:
checked ?

Reply:
If speak about the comment, I forgot to remove it but it works